configuration-management.yaml

```diff
@@ -0,0 +1,302 @@
+openapi: "3.0.0"
+info:
+  description: "This is the Configuration API for UC4."
+  version: "0.10.1"
+  title: "UC4"
+
+servers:
+  - url: https://uc4.cs.upb.de/api/production/configuration-management
+
+tags:
+- name: "Version Management"
+  description: "Everything about the Version"
+  externalDocs:
+    description: "Find out more"
+    url: "https://github.com/upb-uc4"
+- name: "Configuration Management"
+  description: "Everything about the configuration"
+  externalDocs:
+    description: "Find out more"
+    url: "https://github.com/upb-uc4"
+
+paths:
+  /version:
+    get:
+      tags:
+      - "Version Management"
+      summary: "Get version"
+      description: "Returns the version of the current service"
+      operationId: "getVersion"
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JsonVersionNumber'
+  /version/hyperledger:
+    get:
+      tags:
+      - "Version Management"
+      summary: "Get versions of the hyperledger api, chaincode and network"
+      description: "Get versions of the hyperledger api, chaincode and network of the underlying hyperledger fabric network"
+      operationId: "getVersionHyperledger"
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HyperledgerVersions'
+              examples:
+                HyperledgerVersionsExample:
+                  $ref: '#/components/examples/HyperledgerVersions'
+  /configuration:
+    get:
+      tags:
+      - "Configuration Management"
+      summary: "Get the current Configuration"
+      description: "Fetch the current Configuration of the system"
+      operationId: "getConfiguration"
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Configuration'
+              examples:
+                ConfigurationExample:
+                  $ref: '#/components/examples/ConfigurationExample'
+    put:
+      tags:
+      - "Configuration Management"
+      summary: "Send the new Configuration to apply to the system"
+      description: "Send the new Configuration that will serve as the new system configuration. Can only be invoked by admins."
+      operationId: "updateConfiguration"
```

Comment:
What happens, if I create a Lecture, then remove lectures from possible course types using this endpoint?
I guess the lecture will stay in the system. What happens when I try to just update the course name? Will the validation fail because lecture is no longer valid, or will it be valid, because I did not change it?
This case is of course analogous to all validations on parts of this configuration

Will this break your backend, if I send null as a list (instead of the empty list [])? Both is valid json.

Reply:
Good points.  After internal discussion, we decided that changing the configuration at runtime is too complex to be supported